XenModel/Actions/Network/CreateSriovAction.cs

```diff
@@ -56,9 +56,31 @@ protected override void Run()
             // Create the new network            
             XenRef<XenAPI.Network> networkRef = XenAPI.Network.create(Session, newNetwork);
 
+            PIF pifOnMaster =null;
             foreach (PIF thePif in selectedPifs)
             {
-                Network_sriov.async_create(Session, thePif.opaque_ref, networkRef);
+                Host host = thePif.Connection.Resolve<XenAPI.Host>(thePif.host);
+                if (host == null)
+                    continue;
+
+                if (host.IsMaster())
+                {
+                    pifOnMaster = thePif;
+                    break;
+                }
+            }
+
+            Connection.ExpectDisruption = true;
+
+            // Enable SR-IOV network on Pool Master
+            RelatedTask = Network_sriov.async_create(Session, pifOnMaster.opaque_ref, networkRef);
+            PollToCompletion(0, 100);
+
+            // Enable SR-IOV network on Pool Slaves
+            selectedPifs.Remove(pifOnMaster);
+            foreach (PIF thePif in selectedPifs)
+            {
+                Network_sriov.create(Session, thePif.opaque_ref, networkRef);

```

Comment:
Why has this been changed to `create from `async_create` ?

Reply:
Because we need to know whether the action is successful. I was wrong before because I thought if we use the async_create we can also know the action status.